src/blob/index/core.rs

```diff
@@ -64,9 +64,18 @@ where
     for<'a> K: Key<'a>,
 {
     const KEY_SIZE: usize = K::LEN as usize;
-    const BPTREE_ENTRY_SIZE: usize =
+    const BTREE_ENTRY_SIZE: usize =
         size_of::<Vec<u8>>() + MemoryAttrs::<K>::KEY_SIZE + size_of::<Vec<RecordHeader>>();
     const RECORD_HEADER_SIZE: usize = size_of::<RecordHeader>() + MemoryAttrs::<K>::KEY_SIZE;
+    // Each node in BTreeMap contains preallocated vectors of 11 values and 12 edges.
+    // Although count of nodes can't be determined without reimplementing insertion algorithm,
+    // we can use approximation of overhead size added per one key
+    const BTREE_SIZE_MULTIPLIER: usize = (
+        size_of::<std::ptr::NonNull<()>>() +      // ptr to parent btree node
+        size_of::<u16>() * 2 +                    // metadata
+        size_of::<std::ptr::NonNull<()>>() * 12   // edges vector
```

Comment:
According to the description in source code, the node is either `LeafNode` or `InternalNode`: https://doc.rust-lang.org/src/alloc/collections/btree/node.rs.html#123-129

`LeafNode` does not have edges. Only `InternalNode` have them. So the should be a separate calculation for `LeafNode` and for `InternalNode`

Reply:
I think it would be good to define a `const fn` that calculates the multiplier and assign its result to a constant.

Also it would be good to declare constants: `BTREE_NODE_CAPACITY = 11` and `BTREE_NODE_EDGES_COUNT = 12` and use them in calculations

Reply:
Number of `LeafNode` can be estimated as `len / BTREE_NODE_CAPACITY`. `InternalNode`'s connects root with leafs, so it number can be estimated as `leaf_node_count / BTREE_NODE_EDGES_COUNT + leaf_node_count / BTREE_NODE_EDGES_COUNT^2 + ... + leaf_node_count / BTREE_NODE_EDGES_COUNT^5`. I think it would be enough to calculate it up to the 5th power